test/test_gradedunitrangesext_contract.jl

```diff
@@ -1,37 +1,33 @@
-using Test: @test_skip
-# TODO: Add this back once `BlockSparseArrays` is set up.
-@test_skip begin
-  using BlockArrays: Block, blocksize
-  using BlockSparseArrays: BlockSparseArray
-  using GradedUnitRanges: gradedrange
-  using SparseArraysBase: densearray
-  using SymmetrySectors: U1
-  using TensorAlgebra: contract
-  using Random: randn!
-  using Test: @test, @testset
+using BlockArrays: Block, blocksize
+using BlockSparseArrays: BlockSparseArray
+using GradedUnitRanges: dual, gradedrange
+using SparseArraysBase: densearray
+using SymmetrySectors: U1
+using TensorAlgebra: contract
+using Random: randn!
+using Test: @test, @testset
 
-  function randn_blockdiagonal(elt::Type, axes::Tuple)
-    a = BlockSparseArray{elt}(axes)
-    blockdiaglength = minimum(blocksize(a))
-    for i in 1:blockdiaglength
-      b = Block(ntuple(Returns(i), ndims(a)))
-      a[b] = randn!(a[b])
-    end
-    return a
+function randn_blockdiagonal(elt::Type, axes::Tuple)
+  a = BlockSparseArray{elt}(axes)
+  blockdiaglength = minimum(blocksize(a))
+  for i in 1:blockdiaglength
+    b = Block(ntuple(Returns(i), ndims(a)))
+    a[b] = randn!(a[b])
   end
+  return a
+end
 
-  const elts = (Float32, Float64, Complex{Float32}, Complex{Float64})
-  @testset "`contract` `BlockSparseArray` (eltype=$elt)" for elt in elts
-    d = gradedrange([U1(0) => 2, U1(1) => 3])
-    a1 = randn_blockdiagonal(elt, (d, d, d))
-    a2 = randn_blockdiagonal(elt, (d, d, d))
-    a_dest, dimnames_dest = contract(a1, (-1, 1, -2), a2, (-1, -2, 2))
-    a1_dense = densearray(a1)
-    a2_dense = densearray(a2)
-    a_dest_dense, dimnames_dest_dense = contract(
-      a1_dense, (-1, 1, -2), a2_dense, (-1, -2, 2)
-    )
-    @test dimnames_dest == dimnames_dest_dense
-    @test a_dest ≈ a_dest_dense
-  end
+const elts = (Float32, Float64, Complex{Float32}, Complex{Float64})
+@testset "`contract` `BlockSparseArray` (eltype=$elt)" for elt in elts
+  d = gradedrange([U1(0) => 2, U1(1) => 3])
+  a1 = randn_blockdiagonal(elt, (d, d, dual(d), dual(d)))
+  a2 = randn_blockdiagonal(elt, (d, d, dual(d), dual(d)))
+  a_dest, dimnames_dest = contract(a1, (1, -1, 2, -3), a2, (2, -1, 1, -2))
```

Comment:
```suggestion
a_dest, dimnames_dest = contract(a1, (1, 3, 2, -2), a2, (2, 3, 1, -1))
```
, just to follow the `ncon` convention (https://arxiv.org/abs/1402.0939). I see the former version was kind of following the `ncon` convention but had the minus signs reversed. That is purely aesthetic but it took me a while to figure out what contraction was being done.

Also it seems like the duals might be off, since the second dimensions of each tensor are being contracted with each other but they are both non-dual.

Reply:
I fixed the sign convention to abide by `ncon` conventions but I got confused when adding an axis. I think the latest version is correct

Reply:
Looks good now, thanks.